python/tank/bootstrap/manager.py

```diff
@@ -202,30 +208,42 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
-    def set_progress_callback(self, callback):
+    def set_progress_callback(self, progress_callback):
         """
-        Specify a method to call whenever progress should be reported back.
+        Sets a function to call whenever progress of the bootstrap should be reported back.
+
+        This function should have the following signature::
```

Comment:
I think we should add a warning box here telling the user that this callback is not always called from the main thread; in the case of the async bootstrap it is called from a worker thread. DCCs where APIs for logging or console output are sensitive to not running in the main thread (which is common) should make sure to route this through a main thread invoker similar to [what we have in the engine](http://developer.shotgunsoftware.com/tk-core/platform.html?highlight=main#sgtk.platform.Engine.execute_in_main_thread).